### PR TITLE
Okta authentication

### PR DESCRIPTION
**Pull request by rd-huerta**
This PR is for our work on card #29.
There are 44 files changed in this PR due to the fact that we had to merge main into our branch. Please focus on the file changes related to our work on Okta authentication. The important files are listed below:

`src/app/callback/page.tsx`
`src/app/layout.tsx`
`src/app/login/page.tsx`
`src/auth/auth.ts`
`src/app/page.tsx`

To test the changes, update your `.env.local` file with the secrets stored in 1password file called: OKTA_ENV_LOCAL_CM_2.0

### CODE REVIEW

**Review by justin-c-anthony — APPROVED** [review]
🚀

File: src/app/layout.tsx [hunk 1]
```diff
@@ -1,20 +1,45 @@
...
+import { Security } from "@okta/okta-react";
+import oktaAuth from "../auth/auth";
+import { useRouter } from "next/navigation";
+import React from "react";
+import OktaAuth from "@okta/okta-auth-js";
+
+export default function RootLayout({
+    children
+}: {
+    children: React.ReactNode;
```
Comment (sbhamani) [inline src/app/layout.tsx:17]: Same as before you can simply import ReactNode instead of all of React

File: src/app/login/page.tsx [hunk 2]
```diff
@@ -0,0 +1,26 @@
+"use client";
+
+import React, { useEffect } from "react";
+import { useOktaAuth } from "@okta/okta-react";
+import { useRouter } from "next/navigation";
+
+const Login: React.FC = () => {
```
Comment (sbhamani) [inline src/app/login/page.tsx:7]: Same as before

File: src/app/page.tsx [hunk 3]
```diff
@@ -1,8 +1,30 @@
+"use client";
+
+import React, { useEffect } from "react";
```
Comment (sbhamani) [inline src/app/page.tsx:3]: Here weirdly enough we don't need to import React at all. It's somehow implicit even if we don't use the FC component

File: src/app/page.tsx [hunk 4]
```diff
@@ -1,8 +1,30 @@
...
+import { useOktaAuth } from "@okta/okta-react";
+
 export default function Page() {
-    return (
+    const isHerokuEnv = process.env.APP_ENV === "development"; // Check if in Heroku environment
```
Comment (sbhamani) [inline src/app/page.tsx:7]: Can get rid of the comment

**Review by sbhamani — CHANGES_REQUESTED** [review]
just a few import optimizations requested

File: src/app/page.tsx (hunk 3 above)
Comment (justin-c-anthony) [reply src/app/page.tsx:3]: i changed the config file to not need to import react everywhere

**Review by justin-c-anthony — APPROVED** [review]
🔥

**Pull request merged**